File: nosecone/nccpp/ncmain.cpp
// Nosecone Calculator
// UI and Error Catching
// v2.1
// Written by Roman Rodriguez

#include <iostream>
#include <algorithm>
#include <string>
#include "nccalcs.hpp"
using namespace std; 

int main() {
    bool leave;
    list<string> valid_names = {"conic", "cn", "parabolic", "pb", "haack", "hs", "power", "ps", "ellipse", "ep", "ogive", "to"};
    cout << "Welcome to the Nosecone Radius Calculator." << endl;
    cout << "Enter 'help' for help, or 'exit' to close the program." << endl;
    cout << "Valid shape name entries:" << endl;
    for (string name : valid_names) {
        cout << name << " ";
    }
    while (true) {
        string shape;
        cout << "\nEnter shape name: "; cin >> shape;
        if (shape == "exit") {
            cout << "Closing program." << endl;
            exit(0);
        } else if (shape == "help") {
            helpmenu();
        } else if (find(valid_names.begin(), valid_names.end(), shape) != valid_names.end()) {
            string rad_init;
            cout << "Enter radius (INCHES ONLY): "; cin >> rad_init;
            if (rad_init == "exit") {
                cout << "Closing program." << endl;
                exit(0);
            } else if (rad_init == "help") {
                helpmenu();
                rad_init = 1;
                leave = true;
            }
            try {
                float rad_o = stof(rad_init);
            }
            catch (...) {
                cout << "PLEASE ENTER A VALID RADIUS." << endl;
                leave = true; 
            }
            if (leave) {
                leave = false;
            } else {
                float rad_o(stof(rad_init));
                float lng_o(rad_o * 10);
                float length(0);
                calculate(shape, rad_o, lng_o, length);
            }
        } else {
            cout << "PLEASE ENTER A VALID SHAPE NAME.";
        }
    }
    return 0;
}

File: nosecone/ncpython/ncmain.py
# Nosecone Calculator
# UI and Error Catching
# v2.1
# Written by Roman Rodriguez

import nccalcs

valid_names = ["conic", "cn", "parabolic", "pb", "haack", "hs", "power", "ps", "ellipse", "ep", "ogive", "to"]
print("Welcome to the Nosecone Radius Calculator.")
print('Enter "help" for help, or "exit" to close program.')
print("Valid shape name entries:")
print(str(valid_names).replace("[", "").replace("]", "").replace("'", ""))

while True:
    shape = input("Enter shape name: ")
    if shape == "exit":
        print("Closing program.")
        exit(0)
    elif shape == "help":
        nccalcs.helpmenu()
    elif shape not in valid_names:
        print("PLEASE ENTER A VALID SHAPE NAME.")
    else:
        rad_o = input("Enter radius (INCHES ONLY): ")
        if rad_o == "exit":
            print("Closing program.")
            exit(0)
        elif rad_o == "help":
            nccalcs.helpmenu()
            rad_o = 1
            leave = True
        try:
            float(rad_o)
        except:
            print("PLEASE ENTER A VALID RADIUS.")
            leave = True
        if leave:
            leave = False
        else:
            rad_o = float(rad_o)
            lng_o = rad_o * 10
            length = 0
            nccalcs.calculate(shape, rad_o, lng_o, length)
